NKL/Oper.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Govereau
+-/
+
+import NKL.NKI
+
+namespace NKL
+
+-- Some experimental definitions of semantics for expresssions
+-- For now, I am abstracting over the types of tensors and
+-- tensor operators.
+
+-- Evaluating expressions results in values, which
+-- may contain tensors and tensor operators
+inductive Value (tensor op : Type) where
```

Comment:
`op` is the type of operator values. So, if I look up "matmul" in the environment, it will return something that I can give to apply. I don't need to know what the something is, so it is abstract.